migrator/migrations/m_112_to_m_113_recreate_groups_bucket/migration.go

```diff
@@ -0,0 +1,157 @@
+package m112tom113
+
+import (
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/log"
+	"github.com/stackrox/rox/migrator/migrations"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	bolt "go.etcd.io/bbolt"
+)
+
+type groupEntry struct {
+	key   []byte
+	value []byte
+}
+
+var (
+	bucketName = []byte("groups2")
+
+	migration = types.Migration{
+		StartingSeqNum: 112,
+		VersionAfter:   &storage.Version{SeqNum: 113},
+		Run: func(databases *types.Databases) error {
+			return recreateGroupsBucket(databases.BoltDB)
+		},
+	}
+)
+
+func init() {
+	migrations.MustRegisterMigration(migration)
+}
+
+func recreateGroupsBucket(db *bolt.DB) error {
+	// Short-circuit if the bucket does not exist.
+	exists, err := checkGroupBucketExists(db)
+	if err != nil {
+		return errors.Wrap(err, "error checking if groups bucket exists")
+	}
+	if !exists {
+		log.WriteToStderr("groups bucket did not exist, hence no re-creation of the groups bucket was done.")
+		return nil
+	}
+
+	groupEntries, err := fetchGroupsBucket(db)
+	if err != nil {
+		return errors.Wrap(err, "error fetching groups to remove")
+	}
+
+	// Drop the bucket.
+	if err := dropGroupsBucket(db); err != nil {
+		return errors.Wrap(err, "error dropping groups bucket")
+	}
+
+	// Create groups bucket and filter out invalid entries.
+	if err := createGroupsBucket(db, groupEntries); err != nil {
+		return errors.Wrap(err, "error recreating groups bucket")
+	}
+
+	return nil
+}
+
+func fetchGroupsBucket(db *bolt.DB) (groupEntries []groupEntry, err error) {
+	err = db.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(bucketName)
+		// We previously checked that the bucket should be available, but still add this check here.
+		if bucket == nil {
+			return nil
+		}
+		return bucket.ForEach(func(k, v []byte) error {
+			groupEntries = append(groupEntries, groupEntry{key: k, value: v})
+			return nil
+		})
+	})
+	if err != nil {
+		return nil, err
+	}
+	return groupEntries, nil
+}
+
+func dropGroupsBucket(db *bolt.DB) error {
+	return db.Update(func(tx *bolt.Tx) error {
+		return tx.DeleteBucket(bucketName)
+	})
+}
+
+func createGroupsBucket(db *bolt.DB, groupEntries []groupEntry) (err error) {
+	err = db.Update(func(tx *bolt.Tx) error {
+		// Explicitly use the CreateBucket here instead of CreateBucketIfNotExists, as we require the bucket to be
+		// previously dropped.
+		bucket, err := tx.CreateBucket(bucketName)
+		if err != nil {
+			return err
+		}
+
+		var putGroupErrs errorhelpers.ErrorList
+		for _, entry := range groupEntries {
+			// By now, we can assume that the key will be a UUID and the value will be the group proto message.
```

Comment:
Is it because there was previously a migration that changed keys to UUID? Could you reference it here?

Reply:
Added the reference to `105_to_106`